### PR TITLE
Set uninteresting username if user's using a special nick

### PR DESCRIPTION
**Pull request by DanielOaks**
Fixes if clients use the nick as username (which feels pretty common, I know I do it in some of my software), with our new restrictions those clients would no longer be able to connect!

### CODE REVIEW

File: irc/handlers.go [hunk 1]
```diff
@@ -2469,7 +2469,13 @@ func userHandler(server *Server, client *Client, msg ircmsg.IrcMessage, rb *Resp
 
 	err := client.SetNames(msg.Params[0], msg.Params[3], false)
 	if err == errInvalidUsername {
-		rb.Add(nil, server.name, ERR_INVALIDUSERNAME, client.t("Malformed username"))
+		// if client's using a unicode nick or something weird, let's just set 'em up with a blank username instead.
```
Comment (slingamn) [inline irc/handlers.go]: "blank" is a bit misleading

File: irc/handlers.go [hunk 2]
```diff
@@ -2469,7 +2469,13 @@ func userHandler(server *Server, client *Client, msg ircmsg.IrcMessage, rb *Resp
...
+		// fixes clients that just use their nick as a username so they can still use the interesting nick
+		if client.preregNick != "" && !isIdent(client.preregNick) {
```
Comment (slingamn) [inline irc/handlers.go]: actually, should we check `client.preregNick == msg.Params[0]` to catch the exact case?

File: irc/handlers.go (hunk 1 above)
Comment (DanielOaks) [reply irc/handlers.go]: oh yeah, I tried to just set a blank one but it didn't wanna work with me, heh. so ended up just using `user`. nice catch

File: irc/handlers.go (hunk 2 above)
Comment (DanielOaks) [reply irc/handlers.go]: hmmm, interesting. yeah we could definitely do something like that. I'm assuming we don't do any modification (casefolding/etc) of `client.preregNick`, if that's the case that should work fine too.

Comment (slingamn) [reply irc/handlers.go]: Correct, we do a straight `client.preregNick = msg.Params[0]`

Comment (DanielOaks) [reply irc/handlers.go]: perfect, awesome

**Pull request merged**